lib/ansible/modules/cloud/amazon/ec2_group.py

```diff
@@ -276,7 +276,13 @@ def main():
     # find the group if present
     group = None
     groups = {}
-    for curGroup in ec2.get_all_security_groups():
+
+    try:
+        security_groups = ec2.get_all_security_groups()
+    except Exception as e:
```

Comment:
Hm, good point. Maybe making exceptions more specific will be a pr for the future since it's used elsewhere in the module anyway.